src/os/bluestore/BlueStore.cc

```diff
@@ -6409,6 +6418,7 @@ void BlueStore::_txc_finish_io(TransContext *txc)
     if (p->state < TransContext::STATE_IO_DONE) {
       dout(20) << __func__ << " " << txc << " blocked by " << &*p << " "
 	       << p->get_state_name() << dendl;
+      txc->is_pipelined_io = true;
```

Comment:
I don't think this does anything... we already set this to true for this txc when we submitted the io in teh PREPARE case above; no need to set it again.

I think what we actually want is to mark is_pipelined_io if any of the txc's ahead of us have/had IO and are in a state <  KV_DONE.